// @chakra-ui/storybook-addon/dist/preset/index.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/preset/index.ts
var preset_exports = {};
__export(preset_exports, {
  config: () => config,
  managerEntries: () => managerEntries,
  webpackFinal: () => webpackFinal
});
module.exports = __toCommonJS(preset_exports);
function config(entry = []) {
  return [
    ...entry,
    require.resolve("@chakra-ui/storybook-addon/preset/decorators")
  ];
}
function managerEntries(entry = []) {
  return [
    ...entry,
    require.resolve("@chakra-ui/storybook-addon/preset/register")
  ];
}
function webpackFinal(config2) {
  config2.module.rules.push({
    test: /\.mjs$/,
    include: /node_modules/,
    type: "javascript/auto"
  });
  return config2;
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  config,
  managerEntries,
  webpackFinal
});
